src/components/ModalPayment/ModalMegaGam/ModalMegaGam.tsx

```diff
@@ -0,0 +1,333 @@
+import React, { useState, useEffect } from 'react';
+import classnames from 'classnames';
+import styles from './styles.module.scss';
+import {
+  useModalPaymentDispatch,
+  ModalPaymentConstants,
+  ModalPaymentTypes,
+  useModalPaymentState,
+} from '../../../utilities/hooks/ModalPaymentContext';
+import { useTranslation } from 'react-i18next';
+import ReactPixel from 'react-facebook-pixel';
+import CampaignProgressBar from '../../MerchantsPage/gam/CampaignProgressBar';
+import type { Campaign } from '../../../utilities/api/types';
+import { SIZE_TYPE } from '../../MerchantsPage/gam/ProgressBar';
+import styled from 'styled-components';
+import { formatCurrency } from '../../../utilities/general/textFormatter';
+import { calculateFeeAmount } from '../../../utilities/general/feeCalculator';
+import FiscalSponsor from '../../MerchantsPage/gam/FiscalSponsor';
+import { Tooltip } from '@material-ui/core';
+import { Help } from '@material-ui/icons';
+
+export interface Props {
+  sellerId: string;
+  sellerName: string;
+  costPerMeal: number;
+  nonProfitLocationId?: string;
+  projectId?: number;
+  campaign?: Campaign;
+}
+
+export const Modal = (props: Props) => {
+  const { t } = useTranslation();
+  const dispatch = useModalPaymentDispatch(null);
+
+  // set initial number of meals to 1
+  const { feesAmount, fees } = useModalPaymentState(null);
+  const [numberOfMeals, setNumberOfMeals] = useState(1);
+  const donationAmountDollars = numberOfMeals * props.costPerMeal;
+
+  const COST_LIMIT_DOLLARS = 10000;
+  const calculateMatchAmount = (
+    targetAmount: number,
+    amountRaised: number,
+    userDonation: number
+  ) => {
+    if (!targetAmount) return 0;
+
+    let excess = amountRaised + userDonation - targetAmount;
+    return excess > 0 ? Math.max(0, userDonation - excess) : userDonation;
+  };
+  // TODO (billy-yuan): Confirm whether mega-gam campaigns in the future will have donation matching.
+  // If so, we need to add a flag such as isMatching to determine whether to calculate a matchAmount
+  const matchAmount = props.campaign
+    ? calculateMatchAmount(
+        props.campaign.target_amount / 100,
+        props.campaign.amount_raised / 100,
+        donationAmountDollars
+      )
+    : 0;
```

Comment:
This seems to be used 3 times within the code - maybe you can abstract this and call the function instead?